core/src/main/java/org/apache/iceberg/deletes/PositionDeleteWriter.java

```diff
@@ -55,15 +56,27 @@ public PositionDeleteWriter(FileAppender<StructLike> appender, FileFormat format
     this.pathSet = CharSequenceSet.empty();
   }
 
+  @Override
+  public void write(PositionDelete<T> positionDelete) throws IOException {
+    pathSet.add(positionDelete.path());
+    appender.add(positionDelete);
```

Comment:
This class makes an assumption similar to the `ClusteredFileWriter` classes. Should we name it `OrderedPositionDeleteWriter` instead?

Reply:
That will definitely be more descriptive. Do you think it will break anyone? This class existed for a while and we reference it in multiple places such as `WriterFactory`. I think either we keep the original name and deprecate old methods here and in `EqualityDeleteWriter` or just drop old methods and rename as needed.